pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -695,91 +647,66 @@ func (r *ReconcileHyperConverged) ensureNetworkAddons(instance *hcov1alpha1.Hype
-
-			switch conditionsv1.ConditionType(condition.Type) {
-			case conditionsv1.ConditionAvailable:
-				foundAvailableCond = true
-				if condition.Status == corev1.ConditionFalse {
-					msg := fmt.Sprintf("%s is not available: %v", component, string(condition.Message))
-					componentNotAvailable(logger, component, msg, r)
-				}
-			case conditionsv1.ConditionProgressing:
-				if condition.Status == corev1.ConditionTrue {
-					logger.Info(fmt.Sprintf("%s is 'Progressing'", component))
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionProgressing,
-						Status:  corev1.ConditionTrue,
-						Reason:  fmt.Sprintf("%sProgressing", component),
-						Message: fmt.Sprintf("%s is progressing: %v", component, string(condition.Message)),
-					})
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionUpgradeable,
-						Status:  corev1.ConditionFalse,
-						Reason:  fmt.Sprintf("%sProgressing", component),
-						Message: fmt.Sprintf("%s is progressing: %v", component, string(condition.Message)),
-					})
-				}
-			case conditionsv1.ConditionDegraded:
-				if condition.Status == corev1.ConditionTrue {
-					logger.Info(fmt.Sprintf("%s is 'Degraded'", component))
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionDegraded,
-						Status:  corev1.ConditionTrue,
-						Reason:  fmt.Sprintf("%sDegraded", component),
-						Message: fmt.Sprintf("%s is degraded: %v", component, string(condition.Message)),
-					})
-				}
+			Reason:  reconcileCompletedMessage,
+			Message: reconcileCompleted,
+		},
+		{
+			Type:    conditionsv1.ConditionDegraded,
+			Status:  corev1.ConditionFalse,
+			Reason:  reconcileCompletedMessage,
+			Message: reconcileCompleted,
+		},
+		{
+			Type:    conditionsv1.ConditionUpgradeable,
+			Status:  corev1.ConditionTrue,
+			Reason:  reconcileCompletedMessage,
+			Message: reconcileCompleted,
+		},
+	} {
+		// If current condition exists, default to desired condition
+		currentCondition := conditionsv1.FindStatusCondition(instance.Status.Conditions, condition.Type)
+		if currentCondition == nil {
+			currentCondition = &condition
+		}
+
+		// If component condition is missing, then it is unknown
+		componentCondition := conditionsv1.FindStatusCondition(componentConditions, condition.Type)
+		if componentCondition == nil {
+			componentCondition = &conditionsv1.Condition{
+				Type:    condition.Type,
+				Status:  corev1.ConditionUnknown,
+				Reason:  fmt.Sprintf("%sMissingCondition", component),
+				Message: fmt.Sprintf("Component %s is not reporting this condition", component),
 			}
 		}
 
-		if !foundAvailableCond {
-			componentNotAvailable(logger, component, `missing "Available" condition`, r)
+		currentStatus := (currentCondition.Status == condition.Status)
+		componentStatus := (componentCondition.Status == condition.Status)
```

Comment:
`Unknown` will override a "negative" condition (e.g. `Available=False` or `Progressing=True`). I guess that if we already know that one operand is unavailable, we better report that than reporting "unknown". So maybe a value comparison is not enough.

Reply:
I've updated it so that the first time we encounter something other than the desired status, we will update it. Subsequently we will only update the status if the component is both 1) not reporting the desired status and 2) it's status is not unknown.